docs/src/rules/prefer-arrow-callback.md

```diff
@@ -110,6 +110,36 @@ someArray.map(function(item) { return this.doSomething(item); }, someObject);
 
 :::
 
+This rule additionally supports TypeScript type syntax. However, it does not autofix callback `functions` that use `this` as a parameter.
+
+Examples of **incorrect** TypeScript code for this rule:
```

Comment:
Let's also add some examples with `allowUnboundThis` and `allowNamedFunctions` options

Reply:
Done!